webapp/src/router/index.js

```diff
@@ -0,0 +1,23 @@
+import Router from "./Router";
+
+export default (routes) => {
+  const router = new Router(routes, document.getElementById("router"));
+
+  document.addEventListener("DOMContentLoaded", (e) => {
+    document.addEventListener("click", ({ target }) => {
+      const targetElement = target.closest("[route]");
+      if (targetElement) {
+        const matchedRoute = router.match(targetElement.getAttribute("route"));
+        history.pushState({}, matchedRoute.name, matchedRoute.path);
+        router.navigate(matchedRoute);
+
+        const pushStateEvent = new Event("pushstate");
+        window.dispatchEvent(pushStateEvent);
+      }
```

Comment:
```suggestion
      const targetElement = target.closest("[route]");
      if (!targetElement) return;
      const matchedRoute = router.match(targetElement.getAttribute("route"));
      history.pushState({}, matchedRoute.name, matchedRoute.path);
      router.navigate(matchedRoute);

      const pushStateEvent = new Event("pushstate");
      window.dispatchEvent(pushStateEvent);
```
앞선 코드들에도 해당되는 내용인데, if문 때문에 코드의 depth가 깊어지지 않도록 주의해주세요!
특히 else 구문은 아예 사용하지 않는 형태로 코드를 작성하면 더 좋습니다.

참고링크
- https://woowacourse.github.io/javable/post/2020-07-29-dont-use-else/
- https://velog.io/@gomjellie/else-%EC%93%B0%EC%A7%80%EB%A7%88